deepspeed/pt/deepspeed_light.py

```diff
@@ -173,6 +175,33 @@ def __init__(self,
             if self.dump_state():
                 print_configuration(self, 'DeepSpeedLight')
 
+    def _open_mpi_check(self, args):
+        # If OMPI exists in evironment assume we are running inside mpirun
+        if 'OMPI_COMM_WORLD_SIZE' in os.environ:
+            logging.info("Discovered we are probably running inside mpirun, "
+                         "attempting to detect world...")
+            os.environ['RANK'] = os.environ['OMPI_COMM_WORLD_RANK']
+            os.environ['WORLD_SIZE'] = os.environ['OMPI_COMM_WORLD_SIZE']
+            args.local_rank = int(os.environ['OMPI_COMM_WORLD_LOCAL_RANK'])
+            os.environ['MASTER_PORT'] = "29500"
+
+            master_addr = None
+            if hasattr(args, 'master_addr'):
+                os.environ['MASTER_ADDR'] = args.master_addr
+            else:
+                assert 'OMPI_MCA_orte_parent_uri' in os.environ, 'Unable to determine master node IP address'
+                # format: OMPI_MCA_orte_parent_uri=1022885888.0;tcp://192.168.0.186,172.16.1.8
+                os.environ['MASTER_ADDR'] = os.environ['OMPI_MCA_orte_parent_uri'].split(
+                    ";")[1].split(',')[0].replace('tcp://',
+                                                  '')
```

Comment:
I am not sure if this is a general format for this var, might not be the best way to discover the master address. Asking the user to pass via args.master_addr is probably better.